ibis_substrait/tests/compiler/test_tpch.py

```diff
@@ -646,10 +644,27 @@ def tpc_h22(customer, orders):
             reason="Aggregates need to be handled differently than they are",
         ),
     ),
-    pytest.param(lazy_fixture("tpc_h09"), marks=ibis5),
-    pytest.param(lazy_fixture("tpc_h09"), marks=ibis4),
+    pytest.param(
+        lazy_fixture("tpc_h09"),
+        marks=ibis_version("<7.0,~=7.1.0", "Field ordering difference"),
```

Comment:
Weirdly, H9 on Ibis `<7` has the _same_ field order as `7.1.*`. Then `7.0.*` and `7.2.*` have the same field order. But `7.2` casts ints before multiplying them with decimals.